yang/frr-ospfd.yang

```diff
@@ -0,0 +1,957 @@
+      leaf abr-type {
+        type enumeration {
+          enum "cisco" {
+            value 1;
+            description
+              "Alternative ABR, Cisco implementation.";
+          }
+          enum "ibm" {
+            value 2;
+            description
+              "Alternative ABR,  IBM implementation.";
+          }
+          enum "shortcut" {
+            value 3;
+            description
+              "Shortcut ABR.";
+          }
+          enum "standard" {
+            value 4;
+            description
+              "Standard behavior (RFC2328).";
+          }
+        }
+      }
+
+      leaf opaque-lsa {
+        type boolean;
+        description
+          "Enable the Opaque-LSA capability (RFC2370)";
+      }
+
+      leaf rfc1583compatibility {
+        type boolean;
+        description
+          "Enable the RFC 1583 compatibility flag.";
+      }
+
+      leaf router-id {
+        type inet:ipv4-address;
+        description
+          "Router-id for the OSPF process.";
+      }
+
+      leaf write-multiplier {
+        type uint8 {
+          range "1..100";
+        }
+        description
+          "Write multiplier.";
+      }
+    }
+  }
+
+  grouping timer-group {
+    container timers {
+      leaf refresh-interval {
+        type uint16 {
+          range "10..1800";
+        }
+        units "seconds";
+        description
+          "The maximum time between distinct originations of any particular
+           LSA, value in units seconds.";
+      }
+
+      leaf lsa-min-arrival {
+        type uint32 {
+          range "0..600000";
+        }
+        units "milliseconds";
+        description
+          "Minimum delay in receiving new version of an LSA.";
+      }
+
+      container throttle {
+        leaf lsa-all {
+          type uint16 {
+            range "0..5000";
+          }
+          units "milliseconds";
+          description
+            "LSA delay between transmissions.";
+        }
+
+        leaf spf {
+          type uint32 {
+            range "0..600000";
+          }
+          units "milliseconds";
+          description
+            "Delay from first change received till SPF calculation.";
+        }
+      }
+    }
+  }
+
+  grouping segment-routing-group {
+    container segment-routing {
+      container global-block {
+        description
+          "Segment Routing Global Block label range.";
+        must "./upper-bound > ./lower-bound";
+        leaf lower-bound {
+          type uint32 {
+            range "0..1048575";
+          }
+        }
+
+        leaf upper-bound {
+          type uint32 {
+            range "0..1048575";
+          }
+        }
+      }
+
+      leaf node-msd {
+        type uint8 {
+          range "1..16";
+        }
+        description
+          "Maximum Stack Depth for this router.";
+      }
+
+      leaf on {
+        type boolean;
+        description
+          "Enable Segment Routing.";
+      }
+
+      list indices {
```

Comment:
Just a small suggestion: `list prefix-sid` like in isis yang model is more relevant as `list indices`

Reply:
I would prefer you change the prefix list name with same wording as for is-is